protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java

```diff
@@ -15,163 +15,172 @@
+  DEPRECATED_DMN_LATEST_DECISION_REQUIREMENTS_BY_ID(46),
   @Deprecated
-  DEPRECATED_DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY,
+  DEPRECATED_DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY(47),
   @Deprecated
-  DEPRECATED_DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION,
+  DEPRECATED_DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION(48),
   @Deprecated
-  DEPRECATED_DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION,
+  DEPRECATED_DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION(49),
 
   // signal subscription
   @Deprecated
-  DEPRECATED_SIGNAL_SUBSCRIPTION_BY_NAME_AND_KEY,
+  DEPRECATED_SIGNAL_SUBSCRIPTION_BY_NAME_AND_KEY(50),
   @Deprecated
-  DEPRECATED_SIGNAL_SUBSCRIPTION_BY_KEY_AND_NAME,
+  DEPRECATED_SIGNAL_SUBSCRIPTION_BY_KEY_AND_NAME(51),
 
   // distribution
-  PENDING_DISTRIBUTION,
-  COMMAND_DISTRIBUTION_RECORD,
-  MESSAGE_STATS,
+  PENDING_DISTRIBUTION(52),
+  COMMAND_DISTRIBUTION_RECORD(53),
+  MESSAGE_STATS(54),
 
-  PROCESS_INSTANCE_KEY_BY_DEFINITION_KEY,
+  PROCESS_INSTANCE_KEY_BY_DEFINITION_KEY(55),
 
   // This was introduces in 8.3 and backported to earlier versions. As this turns out to not be safe
   // to do we have removed the usage of this CF. We must keep it to remain backwards compatible.
   @Deprecated
-  MIGRATIONS_STATE,
+  MIGRATIONS_STATE(56),
 
-  PROCESS_VERSION,
-  PROCESS_CACHE,
-  PROCESS_CACHE_BY_ID_AND_VERSION,
-  PROCESS_CACHE_DIGEST_BY_ID,
+  PROCESS_VERSION(57),
+  PROCESS_CACHE(58),
+  PROCESS_CACHE_BY_ID_AND_VERSION(59),
+  PROCESS_CACHE_DIGEST_BY_ID(60),
 
-  DMN_DECISIONS,
-  DMN_DECISION_REQUIREMENTS,
-  DMN_LATEST_DECISION_BY_ID,
-  DMN_LATEST_DECISION_REQUIREMENTS_BY_ID,
-  DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY,
-  DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION,
-  DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION,
+  DMN_DECISIONS(61),
+  DMN_DECISION_REQUIREMENTS(62),
+  DMN_LATEST_DECISION_BY_ID(63),
+  DMN_LATEST_DECISION_REQUIREMENTS_BY_ID(64),
+  DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY(65),
+  DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION(66),
+  DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION(67),
 
-  FORMS,
-  FORM_VERSION,
-  FORM_BY_ID_AND_VERSION,
+  FORMS(68),
+  FORM_VERSION(69),
+  FORM_BY_ID_AND_VERSION(70),
 
-  MESSAGES,
-  MESSAGE_START_EVENT_SUBSCRIPTION_BY_NAME_AND_KEY,
-  MESSAGE_START_EVENT_SUBSCRIPTION_BY_KEY_AND_NAME,
-  MESSAGE_SUBSCRIPTION_BY_NAME_AND_CORRELATION_KEY,
-  PROCESS_SUBSCRIPTION_BY_KEY,
+  MESSAGES(71),
+  MESSAGE_START_EVENT_SUBSCRIPTION_BY_NAME_AND_KEY(72),
+  MESSAGE_START_EVENT_SUBSCRIPTION_BY_KEY_AND_NAME(73),
+  MESSAGE_SUBSCRIPTION_BY_NAME_AND_CORRELATION_KEY(74),
+  PROCESS_SUBSCRIPTION_BY_KEY(75),
 
-  JOB_ACTIVATABLE,
+  JOB_ACTIVATABLE(76),
 
-  SIGNAL_SUBSCRIPTION_BY_NAME_AND_KEY,
-  SIGNAL_SUBSCRIPTION_BY_KEY_AND_NAME,
+  SIGNAL_SUBSCRIPTION_BY_NAME_AND_KEY(77),
+  SIGNAL_SUBSCRIPTION_BY_KEY_AND_NAME(78),
 
-  USER_TASKS,
-  USER_TASK_STATES,
```

Comment:
❓ 🤔 I somehow thought/expected that we backport all CFs now, for example https://github.com/camunda/zeebe/blob/main/protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java#L175 is missing.

Reply:
I considered this, but the problem is that it would require fixing the bug in one go on 8.1.

If we would just copy the current CFs file to 8.1, then we also need to add the recovery migration task. Otherwise we're now losing the data on 8.1.

I wanted to split it into two steps:
- first add some safety guards
- then add the migration task to the older versions

Reply:
I think this was not clear to me.

Reply:
Does it make sense or do you think I should change the approach and do it all at once?

Reply:
I think we can also do it like this, but should not forget to do it.